modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2395,6 +2395,14 @@ public CodegenOperation fromOperation(String path,
             if (methodResponse != null) {
                 Schema responseSchema = ModelUtils.unaliasSchema(this.openAPI, ModelUtils.getSchemaFromResponse(methodResponse));
 
+                for (CodegenResponse response : op.responses){
+                    Schema thisResponseSchema = (Schema) response.schema;
+                    if (thisResponseSchema != null) {
+                        CodegenProperty cm = fromProperty("response", thisResponseSchema);
+                        response.dataType = cm.dataType;
```

Comment:
@rienafairefr why are you setting the response's dataType equal to the CodegenProperty dataType here? What is incorrect about how fromResponse is setting response.dataType?